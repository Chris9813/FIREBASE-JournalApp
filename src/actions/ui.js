import { types } from "../types/types";

export const setError = (err) => ({
    type: types.uiSetError,
    payload: err,
})

export const remError = (err) => ({
    type: types.uiRemoveError,
})

