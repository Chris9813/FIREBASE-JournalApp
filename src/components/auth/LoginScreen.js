import React from 'react'


import { useDispatch, useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import { startGoogleLogin, startLoginEmailPaswword } from '../../actions/auth'
import { useForm } from '../hooks/useForm'


export const LoginScreen = () => {

    const {loading} = useSelector(state => state.ui)
    
    const dispatch = useDispatch();
    
    
    const [formValues, handleInputChange ] = useForm({
        email: "chris9813@gmail.com",
        password: "123456"
    });

    const {email, password} = formValues

    const handleLogin = (e) => {
        e.preventDefault()
        dispatch( startLoginEmailPaswword(email, password ));
    }

    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin());
    }


    return (
        <div className="auth_main">
            <div className="auth_box-container">
            <h3 className ="auth_title">Login</h3>
            <form onSubmit={handleLogin}
            className= "animate__animated animate__fadeIn animate__faster"
            >
                <input
                className="aut_input"
                type="text"
                placeholder="email"
                name="email"
                autoComplete="off"
                value = {email}
                onChange={handleInputChange}
                />

                <input
                className="aut_input"
                type="password"
                placeholder="password"
                name="password"
                value={password}
                onChange={handleInputChange}
                />

                <button
                type="submit"
                className = "btn btn-primary btn-block"
                disabled={loading}
                >
                    Login
                </button>


        <div className="auth_social-networks">
            <p>Login with social networks</p>
            
            
            <div 
                className="google-btn"
                onClick= {handleGoogleLogin}
            >
                <div className="google-icon-wrapper">
                    <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                </div>
                <p className="btn-text">
                    <b>Sign in with google</b>
                </p>
            </div>


        </div>
        <Link 
        className="link"
        to = "/auth/register">
            create new account
        </Link>
        </form>
            </div>
            
    </div>
    )
}
