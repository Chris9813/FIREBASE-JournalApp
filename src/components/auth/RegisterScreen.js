import React from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../hooks/useForm'
import validator from "validator"
import { useDispatch, useSelector } from 'react-redux'
import { remError, setError } from '../../actions/ui'
import { startRegisterWithEmailPasswordName } from '../../actions/auth'

export const RegisterScreen = () => {

    const dispatch = useDispatch();

    const {msgError} = useSelector(state => state.ui)


    const[values, handleInputChange] = useForm({
        name: "Chris",
        email: "chris9813@gmail.com",
        password: "123456",
        password2: "123456",
    })
    
    const {name, email, password, password2} = values

    const handleRegister = (e) => {
        e.preventDefault();
        if(isFormValid()) {
            dispatch(startRegisterWithEmailPasswordName(email, password, name))
        }
    }

    const isFormValid = () => {
            if(name.trim().length === 0){
                dispatch(setError("Name is Required"))
                return false
            } else if (!validator.isEmail(email)) {
                dispatch(setError("Email Incorrect"))
                return false
            } else if (password !== password2 || password.length < 5) {
                dispatch(setError("password Incorrect"))
                return false
            }
        dispatch(remError(""))
        return true
    }


    return (
        <div className="auth_main">
            <div className="auth_box-container">
            <div className= "animate__animated animate__fadeIn animate__faster">
            <h3 className ="auth_title">Register</h3>
            <form onSubmit = {handleRegister}>

            {
            msgError && 
            (<div className="auth_alert-error">
            {msgError}
            </div>)
            }

            <input
                className="aut_input"
                type="text"
                placeholder="Email"
                name="email"
                autoComplete="off"
                onChange={handleInputChange}
                value={email}
                />

                <input
                className="aut_input"
                type="text"
                placeholder="Name"
                name="name"
                autoComplete="off"
                onChange={handleInputChange}
                value={name}
                />

                <input
                className="aut_input"
                type="password"
                placeholder="password"
                name="password"
                onChange={handleInputChange}
                value={password}
                />

                <input
                className="aut_input"
                type="password"
                placeholder="password2"
                name="password2"
                onChange={handleInputChange}
                value={password2}
                />

                <button
                type="submit"
                className = "btn btn-primary btn-block"
                onClick={handleRegister}
                >
                    Register
                </button>
        <Link 
        className="link"
        to = "/auth/login">
            Already register
        </Link>
        </form>
    </div>
            </div>
            
        </div>
    )
}
