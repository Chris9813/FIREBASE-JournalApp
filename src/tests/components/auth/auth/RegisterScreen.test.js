import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, mount } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { startRegisterWithEmailPasswordName } from "../../../../actions/auth";
import { LoginScreen } from "../../../../components/auth/LoginScreen";
import { RegisterScreen } from "../../../../components/auth/RegisterScreen";
import { types } from "../../../../types/types";



const middlewares = [thunk];
const mockStore = configureStore(middlewares);

configure({ adapter: new Adapter() });

const initState = {
    auth: { },
    ui: {
        loading: false,
        msgError: null,
    }
};

let store = mockStore(initState)
//store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <RegisterScreen />
        </MemoryRouter>
    </Provider>
    );

describe('Pruebas en <RegisterScreen />', () => {
    test('debe de mostrrse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de hacer dispatch de la accion respectiva ', () => {
        const emailField = wrapper.find('input[name="email"]')
        emailField.simulate("change", {
            target: {
                value: '',
                name: 'email',
            }
        });
        wrapper.find("form").simulate("submit", {
            preventDefault(){}
        });
        const actions = store.getActions();
        expect(actions[0]).toEqual({
            type: types.uiSetError,
            payload: 'Email Incorrect',
        });
    });
    
    test('debe dde mostrar alert de error ', () => {
        
        const initState = {
            auth: { },
            ui: {
                loading: false,
                msgError: 'Email Incorrect',
            }
        };
        
        const store = mockStore(initState)
        
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <RegisterScreen />
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper.find('.auth_alert-error').text().trim()).toBe('Email Incorrect');

    });
    

});
