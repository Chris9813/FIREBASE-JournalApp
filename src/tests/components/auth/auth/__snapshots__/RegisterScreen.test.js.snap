// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RegisterScreen /> debe de mostrrse correctamente  1`] = `
<div
  className="auth_main"
>
  <div
    className="auth_box-container"
  >
    <div
      className="animate__animated animate__fadeIn animate__faster"
    >
      <h3
        className="auth_title"
      >
        Register
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="aut_input"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value="chris9813@gmail.com"
        />
        <input
          autoComplete="off"
          className="aut_input"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value="Chris"
        />
        <input
          className="aut_input"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value="123456"
        />
        <input
          className="aut_input"
          name="password2"
          onChange={[Function]}
          placeholder="password2"
          type="password"
          value="123456"
        />
        <button
          className="btn btn-primary btn-block"
          onClick={[Function]}
          type="submit"
        >
          Register
        </button>
        <a
          className="link"
          href="/auth/login"
          onClick={[Function]}
        >
          Already register
        </a>
      </form>
    </div>
  </div>
</div>
`;
